install.sh

```diff
@@ -94,133 +103,128 @@ esac
 
 # functions to maintain packages
 function download_check () {
-    cd $CREW_BREW_DIR
+    cd "${CREW_BREW_DIR}"
 
     #download
-    echo "Downloading $1..."
-    curl --progress-bar -C - -# -L --ssl $2 -o "$3"
+    echo "Downloading ${1}..."
+    curl --progress-bar -C - -L --ssl "${2}" -o "${3}"
 
     #verify
-    echo "Verifying $1..."
-    echo $4 $3 | sha256sum -c -
-    case $? in
+    echo "Verifying ${1}..."
+    echo "${4}" "${3}" | sha256sum -c -
+    case "${?}" in
     0) ;;
     *)
-      echo "Verification failed, something may be wrong with the $1 download."
+      echo "Verification failed, something may be wrong with the download."
       exit 1;;
     esac
 }
 
 function extract_install () {
-    cd $CREW_BREW_DIR
+    cd "${CREW_BREW_DIR}/dest"
 
     #extract and install
-    echo "Extracting $1 (this may take a while)..."
-    rm -rf ./usr ./home
-    tar -xf $2
-    echo "Installing $1 (this may take a while)..."
-    if [ -d "${CREW_BREW_DIR}/home" ]; then
-      tar cf - ./usr/* ./home/* | (cd /; tar xp --keep-directory-symlink -f -)
-    else
-      tar cf - ./usr/* | (cd /; tar xp --keep-directory-symlink -f -)
-    fi
-    mv ./dlist $CREW_CONFIG_PATH/meta/$1.directorylist
-    mv ./filelist $CREW_CONFIG_PATH/meta/$1.filelist
+    echo "Extracting ${1} (this may take a while)..."
+    for dir in ./*/; do
+      rm -rf "${dir}"
+    done
+    for file in ./*list; do
+      rm -f "${file}"
+    done
+    tar xpf ../"${2}"
+    echo "Installing ${1} (this may take a while)..."
+    tar cpf - ./*/* | (cd /; tar xp --keep-directory-symlink -f -)
+    mv ./dlist "${CREW_CONFIG_PATH}/meta/${1}.directorylist"
+    mv ./filelist "${CREW_CONFIG_PATH}/meta/${1}.filelist"
 }
 
 function update_device_json () {
-  cd $CREW_CONFIG_PATH
+  cd "${CREW_CONFIG_PATH}"
 
-  if grep '"name": "'$1'"' device.json > /dev/null; then
-    echo "Updating version number of existing $1 information in device.json..."
-    sed -i device.json -e '/"name": "'$1'"/N;//s/"version": ".*"/"version": "'$2'"/'
+  if grep '"name": "'"${1}"'"' device.json > /dev/null; then
+    echo "Updating version number of ${1} in device.json..."
+    sed -i device.json -e '/"name": "'"${1}"'"/N;//s/"version": ".*"/"version": "'"${2}"'"/'
   elif grep '^    }$' device.json > /dev/null; then
-    echo "Adding new $1 information to device.json..."
+    echo "Adding new information on ${1} to device.json..."
```

Comment:
How about?
```
echo "Adding new information for package '${1}' to device.json..."
```
This makes sure if the package name is an ordinary english word, there is no confusion.

Reply:
I don't think `libssh2` or `libiconv` are ordinary English words.

Reply:
Then why change it?

Reply:
Sounds perfectly fine to me.

Reply:
"Sounds" doesn't mean it's correct.

Reply:
I'll go for a compromise: `    echo "Adding new information on package '${1}' to device.json..."`

Reply:
Actually, `for` is a better preposition than `on` here.  Didn't you learn these things in elementary school?

Reply:
I don't appreciate being insulted, @uberhacker. Please look at the general channel in Slack before you post things like this.